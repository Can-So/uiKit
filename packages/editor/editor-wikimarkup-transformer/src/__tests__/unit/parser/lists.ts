import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Lists', () => {
  const testCases: Array<[string, string]> = [
    [
      'should find unordered lists where items start with *',
      `* some
* bullet
** *indented*
** bullets
* points`,
    ],
    [
      'should find unordered lists where items start with -',
      `- different
- bullet
- types`,
    ],
    [
      'should find ordered lists',
      `# a
# numbered
# list`,
    ],
    [
      'should find mixed ordered -> unordered lists',
      `# a
# numbered
#* with
#* nested
#* bullet
# list`,
    ],
    [
      'should find mixed unordered -> ordered lists',
      `* a
* bulleted
*# with
*# nested
*# numbered
* list`,
    ],
    ['should convert blockquotes into paragraphs', '* bq. This is sparta'],
    ['should convert headings into paragraphs', '* h1. Foo'],
    ['should convert rules into paragraphs', '* ----'],
    ['should use mediaSingle nodes', '* !attached-image.gif!'],
    [
      'should not fail for multiple lists',
      // tslint:disable-next-line:max-line-length
      "Steps to reproduce:\r\n* Type something\r\n* Type @\r\n* Type few letters\r\n* mention mark disappears \r\n\r\nExpected:\r\n* Mention mark doesn't disappear",
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
