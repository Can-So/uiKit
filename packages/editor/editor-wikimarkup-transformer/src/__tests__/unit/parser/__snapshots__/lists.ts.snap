// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Lists should be able to create nested list with - 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "different",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "bullet",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "types",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should break out the media group 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "This is a media group ",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "__key": null,
            "collection": "",
            "height": null,
            "id": "file.pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": " and text after it",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should break out the media group even in nested list 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "nested",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "media group ",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "__key": null,
            "collection": "",
            "height": null,
            "id": "file.pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": " here",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "nice nice",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "all good",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should convert blockquotes into paragraphs 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "This is sparta",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should convert headings into paragraphs 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "marks": Array [
                    Object {
                      "type": "strong",
                    },
                  ],
                  "text": "FOO",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should convert rules into paragraphs 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "----",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should find mixed ordered -> unordered lists 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "numbered",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "with",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "nested",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "bullet",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "list",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should find mixed unordered -> ordered lists 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "bulleted",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "with",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "nested",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "numbered",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "orderedList",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "list",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should find ordered lists 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "a",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "numbered",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "list",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should find unordered lists where items start with * 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "some",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "bullet",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "marks": Array [
                            Object {
                              "type": "strong",
                            },
                          ],
                          "text": "indented",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "bullets",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "points",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should find unordered lists where items start with - 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "different",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "bullet",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "types",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should not fail for multiple lists 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Steps to reproduce:",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Type something",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Type @",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Type few letters",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "mention mark disappears ",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
    Object {
      "content": Array [
        Object {
          "text": "Expected:",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Mention mark doesn't disappear",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Lists should use mediaSingle nodes 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "layout": "center",
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__displayType": null,
                    "__fileMimeType": null,
                    "__fileName": null,
                    "__fileSize": null,
                    "__key": null,
                    "collection": "",
                    "height": 183,
                    "id": "attached-image.gif",
                    "occurrenceKey": null,
                    "type": "file",
                    "url": null,
                    "width": 200,
                  },
                  "type": "media",
                },
              ],
              "type": "mediaSingle",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;
